bindings/commercetools/commercetools.go

```diff
@@ -0,0 +1,196 @@
+func (ct *Binding) Init(metadata bindings.Metadata) error {
+	commercetoolsM, err := ct.getCommercetoolsMetadata(metadata)
+	if err != nil {
+		return err
+	}
+
+	// Create the new client. When an empty value is passed it will use the CTP_*
+	// environment variables to get the value. The HTTPClient arg is optional,
+	// and when empty will automatically be created using the env values.
+	client, err := commercetools.NewClient(&commercetools.ClientConfig{
+		ProjectKey: commercetoolsM.projectKey,
+		Endpoints:  commercetools.NewClientEndpoints(commercetoolsM.region, commercetoolsM.provider),
+		Credentials: &commercetools.ClientCredentials{
+			ClientID:     commercetoolsM.clientID,
+			ClientSecret: commercetoolsM.clientSecret,
+			Scopes:       []string{commercetoolsM.scopes},
+		},
+	})
+	if err != nil {
+		ct.logger.Errorf("error creating commercetools client: %s", err)
+
+		return err
+	}
+
+	ct.client = client
+
+	return nil
+}
+
+func (ct *Binding) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{bindings.CreateOperation}
+}
+
+// Invoke is triggered from Dapr.
+func (ct *Binding) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	var reqData Data
+	json.Unmarshal(req.Data, &reqData)
+	query := reqData.Query
+
+	ctx := context.Background()
+
+	res := &bindings.InvokeResponse{Data: nil, Metadata: nil}
+	var err error
+
+	if len(reqData.CommercetoolsAPI) > 0 {
+		ct.logger.Infof("commercetoolsAPI: %s", reqData.CommercetoolsAPI)
+		if reqData.CommercetoolsAPI == "GraphQLQuery" {
+			res, err = handleGraphQLQuery(ctx, ct, query)
+			if err != nil {
+				ct.logger.Errorf("error GraphQLQuery")
+
+				return nil, err
+			}
+		}
+	} else {
+		return nil, errors.New("commercetools error: No commercetools API provided")
+	}
+
+	return res, nil
+}
+
+// HandleGraphQLQuery executes the provided query against the commercetools backend.
+func handleGraphQLQuery(ctx context.Context, ct *Binding, query string) (*bindings.InvokeResponse, error) {
+	ct.logger.Infof("handleGraphQLQuery")
+
+	res := &bindings.InvokeResponse{Data: nil, Metadata: nil}
+
+	if len(query) > 0 {
+		gql := ct.client.NewGraphQLQuery(query)
+		var gqlResp interface{}
+		errGQL := gql.Execute(&gqlResp)
+		if errGQL != nil {
+			return nil, errors.New("commercetools error: Error executing the provided GraphQL query")
+		}
+
+		bQuery, errM := json.Marshal(gqlResp)
+		if errM != nil {
+			return nil, errors.New("commercetools error: Error marshalling GraphQL query result")
+		}
+
+		res = &bindings.InvokeResponse{Data: bQuery, Metadata: nil}
+	} else {
+		return res, errors.New("commercetools error: No GraphQL query is provided")
+	}
+
+	return res, nil
+}
+
+// getCommercetoolsMetadata returns new commercetools metadata.
+func (ct *Binding) getCommercetoolsMetadata(metadata bindings.Metadata) (*commercetoolsMetadata, error) {
+	meta := commercetoolsMetadata{}
+
+	val, ok := metadata.Properties["region"]
```

Comment:
Suggestion: it's a fairly common pattern in the other components to collapse the two error cases into one, so feel free to do so:
```go
if val, ok := metadata.Properties["region"]; ok && val != "" {
	meta.region= val
} else {
	return nil, errors.New("commercetools error: missing `region` configuration")
}
```

Reply:
Followed your suggestion and collapsed all these separate error cases into one